src/core/reference/include/openvino/reference/concat.hpp

```diff
@@ -7,22 +7,16 @@
 #include <vector>
 
 #include "openvino/core/shape.hpp"
+#include "openvino/core/type/element_type.hpp"
 
 namespace ov {
 namespace reference {
-void concat(const std::vector<const char*>& args,
-            char* out,
+void concat(const std::vector<const void*>& args,
```

Comment:
It would be better to use `char*` to be aligned with others which works on byte like memories instead OV types. 
If there is additional information about type it can be casted to that type, also pointer arithmetic on `void*` is not allowed and this is common use in reference implementations.

Reply:
Tensor `.data()` returns `void*` anyway, so previous proposal just postponed the moment of `static_cast`.
Also pointer arithmetic on `void*` was not performed and needed for Concat.

Anyway as discussed, usage of `void*` has been reverted to keep `char*` and `reinterpret_cast` to `string*`.
Additionally default `element::Type_t::undefined` has been proposed to keep current api of Concat reference, and avoid several changes in other files, where reference::concat is already used.